renderlib/FileReaderTIFF.cpp

```diff
@@ -317,21 +321,56 @@ readTiffDimensions(TIFF* tiff, const std::string filepath, VolumeDimensions& dim
 
 // return number of bytes copied to dest
 size_t
-copyAndConvert(uint8_t* dest, const uint8_t* src, size_t numBytes, int srcBitsPerPixel)
+copyDirect(uint8_t* dest, const uint8_t* src, size_t numBytes, int srcBitsPerPixel)
+{
+  memcpy(dest, src, numBytes);
+  return numBytes;
+}
+
+// convert pixels
+// this assumes tight packing of pixels in both buf(source) and dataptr(dest)
+// assumes dest is of format IN_MEMORY_BPP
+// return 1 for successful conversion, 0 on failure (e.g. unacceptable srcBitsPerPixel)
+size_t
+convertChannelData(uint8_t* dest, const uint8_t* src, const VolumeDimensions& dims)
 {
+  // how many pixels in this channel:
+  size_t numPixels = dims.sizeX * dims.sizeY * dims.sizeZ;
+  int srcBitsPerPixel = dims.bitsPerPixel;
+
   // dest bits per pixel is IN_MEMORY_BPP which is currently 16, or 2 bytes
   if (IN_MEMORY_BPP == srcBitsPerPixel) {
-    memcpy(dest, src, numBytes);
-    return numBytes;
+    memcpy(dest, src, numPixels * (srcBitsPerPixel / 8));
+    return 1;
   } else if (srcBitsPerPixel == 8) {
-    // convert pixels
-    // this assumes tight packing of pixels in both buf(source) and dataptr(dest)
     uint16_t* dataptr16 = reinterpret_cast<uint16_t*>(dest);
-    for (size_t b = 0; b < numBytes; ++b) {
+    for (size_t b = 0; b < numPixels; ++b) {
       *dataptr16 = (uint16_t)src[b];
       dataptr16++;
     }
-    return numBytes * 2;
+    return 1;
+  } else if (srcBitsPerPixel == 32) {
+    // assumes 32-bit floating point (not int or uint)
+    uint16_t* dataptr16 = reinterpret_cast<uint16_t*>(dest);
```

Comment:
Why are you compressing your numerical image range? 
Is this to save space or ? 

I suspect there's a reason you are using uint16 throughout your types but it's not clear why, maybe add a comment near here somewhere?

Reply:
This function is converting everything to 16-bit uints. This code branch is converting 32-bit floats (src) to 16-bit uints (dest).